### PR TITLE
Navigation bar color adaption, background notification update, and announcement hiding

### PR DESCRIPTION
**Pull request by BertrandTheHealer**
Add option to change navigation bar color to match:
* Theme (if theme used) on game pages
* First panel (if no theme) on game pages
* User's avatar on profile pages

Add option to update game completion/comment/emote notifications periodically and optionally display desktop notifications when they arrive

Add option to hide palette/contest/featured game announcement banner on home page.

### CODE REVIEW

File: drawception-anbt.user.js [hunk 1]
```diff
@@ -2244,6 +2247,9 @@ const wrapped = () => {
...
     }
+    if (options.hideAnnouncements) {
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: Who wants to hide announcements?

File: drawception-anbt.user.js [hunk 2]
```diff
@@ -2314,6 +2320,254 @@ const wrapped = () => {
...
+  function launchNotifications(notification, unreadNotifications) {
+    $("#btn-notifications").click();//open unread menu
+  }
+
+  //check for notifications every 30 seconds
+  if(options.checkForNotifications){var notificationTimer = setInterval(checkNotifications, 30000);}
+
+  function checkNotifications(){
+      //load new notification number
+      var unreadCount = "";
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: Avoid var, use const/let instead...

File: drawception-anbt.user.js [hunk 3]
```diff
@@ -2314,6 +2320,254 @@ const wrapped = () => {
...
+      var xhttp = new XMLHttpRequest();
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: Another request? Not very good with bad connection...

File: drawception-anbt.user.js [hunk 4]
```diff
@@ -2314,6 +2320,254 @@ const wrapped = () => {
...
+  try{
+    if(options.checkForNotifications){
+      Notification.requestPermission();//get notification permissions if they aren't already granted
+      checkNotifications();//check for new notifications
+    }
+  }catch(error){}
+
+  //runs when user clicks on notification
+  function launchNotifications(notification, unreadNotifications) {
+    $("#btn-notifications").click();//open unread menu
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: .click() comes with JQuery, no?

File: drawception-anbt.user.js [hunk 5]
```diff
@@ -2314,6 +2320,254 @@ const wrapped = () => {
...
+      }
+      $("#nav-drag").style.backgroundSize = ".5px .5px";
+      $("#nav-drag").style.backgroundRepeat = "repeat";
+    }catch(error){}
+
+    //set navbar color based on theme
+    try{//if the game is not a vet game the theme will be in the first "label-no-select" element
+      var navbarColor = "#0CE853";
+      var theme = document.getElementsByClassName("label-big")[0].firstChild.innerHTML;
+      switch(theme) {
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: 2 switches? I'm not a big fan of that...

File: drawception-anbt.user.js [hunk 6]
```diff
@@ -2314,6 +2320,254 @@ const wrapped = () => {
...
+          break;
+        default:
+          //leave it green
+      }
+      if(navbarColor != "#0CE853"){
+        //if there is a theme remove the color generated from the first drawing
+        $("#nav-drag").style.background = "";
+      }
+      $("#nav-drag").style.backgroundColor = navbarColor;
+    }catch(error){}
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: You catch the error, but you do anything with it... Not very good...

File: drawception-anbt.user.js [hunk 7]
```diff
@@ -2314,6 +2320,254 @@ const wrapped = () => {
...
+    try{//if the game is a vet game the theme will be in the second "label-no-select" element
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: This is useless... Reuse the code of the previous try...

File: drawception-anbt.user.js [hunk 8]
```diff
@@ -2314,6 +2320,254 @@ const wrapped = () => {
...
+  function checkNotifications(){
+      //load new notification number
+      var unreadCount = "";
+      var xhttp = new XMLHttpRequest();
+      xhttp.onreadystatechange = function() {
+       if (this.readyState == 4 && this.status == 200) {
+         parser = new DOMParser();
+         doc = parser.parseFromString(xhttp.responseText, "text/html");//parse the homepage
+         unreadCount = doc.getElementsByTagName("btn-notifications")[0].getAttribute(":amount");//doc.getElementById("user-notify-count").innerHTML;//get unread count
+         console.log(unreadCount+" new notifications");
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: This console.log is for debugging... Useless to keep it...

File: drawception-anbt.user.js [hunk 9]
```diff
@@ -2314,6 +2320,254 @@ const wrapped = () => {
...
+    }
+  }catch(error){}
+
+  //runs when user clicks on notification
+  function launchNotifications(notification, unreadNotifications) {
+    $("#btn-notifications").click();//open unread menu
+  }
+
+  //check for notifications every 30 seconds
+  if(options.checkForNotifications){var notificationTimer = setInterval(checkNotifications, 30000);}
```
Comment (EnderDragonneau) [inline drawception-anbt.user.js]: A request in an interval of 30s ? A nigthmare with bad connection... I can't add that...

**Comment by BertrandTheHealer** [comment]
My code was a lot sloppier than I thought. I'll clean it up and submit the features in separate PRs.

File: drawception-anbt.user.js (hunk 1 above)
Comment (BertrandTheHealer) [reply drawception-anbt.user.js]: Me

File: drawception-anbt.user.js (hunk 5 above)
Comment (BertrandTheHealer) [reply drawception-anbt.user.js]: You're right, there's a better way to do it.

File: drawception-anbt.user.js (hunk 7 above)
Comment (BertrandTheHealer) [reply drawception-anbt.user.js]: Agreed

File: drawception-anbt.user.js (hunk 8 above)
Comment (BertrandTheHealer) [reply drawception-anbt.user.js]: Yup. Left it in by accident.

File: drawception-anbt.user.js (hunk 9 above)
Comment (BertrandTheHealer) [reply drawception-anbt.user.js]: Definitely, perhaps change to 5 minutes?

Comment (EnderDragonneau) [reply drawception-anbt.user.js]: It is better to use the least possible requests... I think it would be better to forget this feature...

**Pull request closed without merging**